pkg/controller/garbagecollector/metaonly/types.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metaonly
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// MetaOnly allows decoding only the apiVersion, kind, and metadata fields of
+// JSON data.
+// TODO: enable meta-only decoding for protobuf.
+type MetaOnly struct {
```

Comment:
I guess we could include that in the swagger if we had to.  We definitely
have to know about the case when the object doesn't have ObjectMeta -
knowing it has tag 1 isn't enough.

On Tue, Aug 2, 2016 at 5:40 PM, Daniel Smith <notifications@github.com>
wrote:

> In pkg/controller/garbagecollector/metaonly/types.go
> <https://github.com/kubernetes/kubernetes/pull/28480#discussion_r73243331>
> :
>
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package metaonly
> > +
> > +import (
> > +	"k8s.io/kubernetes/pkg/api/unversioned"
> > +	"k8s.io/kubernetes/pkg/api/v1"
> > +)
> > +
> > +// MetaOnly allows decoding only the apiVersion, kind, and metadata fields of
> > +// JSON data.
> > +// TODO: enable meta-only decoding for protobuf.
> > +type MetaOnly struct {
>
> Currently all object metas are tag 1 in our code base. Can we force this
> as a rule?
>
> I think this is sort of the crux of the matter. I think it would be super
> helpful if we could do this, not just for this purpose, but for others, too
> (imagine pushing indexing down into the storage layer).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28480/files/41d92b1b3fe4883311ae3037fb816f858d565fbd#r73243331>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p0-HdvcP1v_2DrZTcMaoFtwZWo_Gks5qb7lUgaJpZM4JEvyT>
> .
>